nvflare/client/api.py

```diff
@@ -191,11 +195,20 @@ def log(key: str, value: Any, data_type: AnalyticsDataType, **kwargs):
     Returns:
         whether the key value pair is logged successfully
     """
-    global client_api
-    return client_api.log(key, value, data_type, **kwargs)
+    global default_context
+    local_ctx = ctx if ctx else default_context
+    return local_ctx.api.log(key, value, data_type, **kwargs)
 
 
-def clear():
+def clear(ctx: Optional[APIContext] = None):
     """Clears the cache."""
-    global client_api
-    return client_api.clear()
+    global default_context
+    local_ctx = ctx if ctx else default_context
+    return local_ctx.api.clear()
+
+
+def shutdown(ctx: Optional[APIContext] = None):
+    """Releases all threads and resources used by the API and stops operation."""
+    global default_context
+    local_ctx = ctx if ctx else default_context
+    return local_ctx.api.shutdown()
```

Comment:
Should check local_ctx is not None. If none, either raise exception or simply pass since nothing to clean.